### PR TITLE
chore: upgrade flutter 3.13.x

### PR DESCRIPTION
**Pull request by LucasXu0**
### Feature Preview

- upgrade to Flutter stable channel to fix some mobile issues.
- adapt iOS 17

---

#### PR Checklist

- [ ] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/base/link_to_page_widget.dart [hunk 1]
```diff
@@ -44,13 +44,15 @@ void showLinkToPageMenu(
...
-              ),
-              context,
-            );
+            if (context.mounted) {
+              Dialogs.show(
+                child: FlowyErrorPage.message(
+                  e.msg,
+                  howToFix: LocaleKeys.errorDialog_howToFixFallback.tr(),
+                ),
+                context,
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/base/link_to_page_widget.dart:53]: I know this was not introduced in this PR, but this seems odd to me, that a named parameter is before a positional, and the named parameter is `child` but isn't sorted last.

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/custom_cover_picker.dart [hunk 2]
```diff
@@ -322,7 +322,7 @@ class _CoverImagePreviewWidgetState extends State<CoverImagePreviewWidget> {
                     (l) => _buildImageDeleteButton(context),
                     (r) => Container(),
                   )
-                : Container()
+                : Container(),
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/custom_cover_picker.dart:325]:
```suggestion
                : const SizedBox.shrink(),
```

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_file_customize_location_view.dart [hunk 3]
```diff
@@ -172,7 +173,10 @@ class _ChangeStoragePathButtonState extends State<_ChangeStoragePathButton> {
         onPressed: () async {
           // pick the new directory and reload app
           final path = await getIt<FilePickerService>().getDirectoryPath();
-          if (path == null || !mounted || widget.usingPath == path) {
+          if (path == null || widget.usingPath == path) {
+            return;
+          }
+          if (!mounted) {
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_file_customize_location_view.dart:179]: Why this change?
```suggestion
          if (!mounted || path == null || widget.usingPath == path) {
```

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_file_customize_location_view.dart [hunk 4]
```diff
@@ -245,7 +249,10 @@ class _RecoverDefaultStorageButtonState
         // reset to the default directory and reload app
         final directory = await appFlowyApplicationDataDirectory();
         final path = directory.path;
-        if (!mounted || widget.usingPath == path) {
+        if (widget.usingPath == path) {
+          return;
+        }
+        if (!mounted) {
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_file_customize_location_view.dart:255]: Why this change?
```suggestion
        if (!mounted || widget.usingPath == path) {
```

File: frontend/scripts/code_generation/freezed/generate_freezed.sh [hunk 5]
```diff
@@ -26,28 +26,28 @@ fi
...
+#     fi
+#     dart run build_runner clean && dart run build_runner build -d
+#     echo "Done running build command in $d"
+#   else
+#     echo "No pubspec.yaml found in $d, it can\'t be a Dart project. Skipping."
+#   fi
+
+#   # Navigate back to the packages directory
+#   cd ..
+# done
```
Comment (Xazin) [inline frontend/scripts/code_generation/freezed/generate_freezed.sh:50]: Why this change?

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_file_customize_location_view.dart (hunk 3 above)
Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_file_customize_location_view.dart:179]: Because Flutter 3.13 will raise an error if we combine the !mounted and other conditions.

File: frontend/scripts/code_generation/freezed/generate_freezed.sh (hunk 5 above)
Comment (LucasXu0) [reply frontend/scripts/code_generation/freezed/generate_freezed.sh:50]: It's unnecessary to generate the freezed files in the packages because we didn't import freezed on them, and this command really takes time.

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/base/link_to_page_widget.dart (hunk 1 above)
Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/base/link_to_page_widget.dart:53]: I know, but I don't want to include this change in this PR since it has a lot of changes that will conflict with the main branch easily.
